src/Microsoft.NetCore.Analyzers/Core/Security/UseAutoValidateAntiforgeryToken.cs

```diff
@@ -215,9 +224,10 @@ public override void Initialize(AnalysisContext context)
                     var derivedControllerTypeSymbol = (INamedTypeSymbol)symbolAnalysisContext.Symbol;
                     var baseTypes = derivedControllerTypeSymbol.GetBaseTypes();
 
-                    // An subtype of `Microsoft.AspNetCore.Mvc.Controller` or `Microsoft.AspNetCore.Mvc.ControllerBase`).
+                    // An subtype of `Microsoft.AspNetCore.Mvc.Controller`, which indicates that cookie-based authentication is used and thus CSRF is a concern.
                     if (baseTypes.Contains(controllerTypeSymbol) ||
-                        baseTypes.Contains(controllerBaseTypeSymbol))
+                            (!onlyLookAtDerivedClassesOfController &&
+                            baseTypes.Contains(controllerBaseTypeSymbol)))
```

Comment:
Looking at the implementation, I think a better option name here might be `exclude_mvc_controller_base`? @dotpaul?

Reply:
Whatever the name is, I think "aspnet_core" ought to appear in the name somewhere.